pkg/targetgroupbinding/networking_manager_test.go

```diff
@@ -1412,3 +1414,368 @@ func Test_defaultNetworkingManager_computeRestrictedIngressPermissionsPerSG(t *t
 		})
 	}
 }
+
+func Test_defaultNetworkingManager_resolveEndpointSGForENI(t *testing.T) {
+	type fetchSGInfosByIDCall struct {
+		req  []string
+		resp map[string]networking.SecurityGroupInfo
+		err  error
+	}
+
+	type fields struct {
+		fetchSGInfosByRequestCalls []fetchSGInfosByIDCall
+		serviceTargetENISGTags     map[string]string
+	}
+	type args struct {
+		ctx     context.Context
+		eniInfo networking.ENIInfo
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    string
+		wantErr bool
```

Comment:
A better pattern is to have `wantErr string` with the value being the expected error message.